newIDE/app/src/VariablesList/index.js

```diff
@@ -254,43 +252,75 @@ export default class VariablesList extends React.Component<Props, State> {
     );
   }
 
-  render() {
+  onAddVariable = (variableName = false,string='') => {
     const { variablesContainer } = this.props;
+    const variable = new gd.Variable();
+    variable.setString(string);
+
+    const name = variableName
+      ? variableName
+      : newNameGenerator('Variable', name => variablesContainer.has(name));
+    variablesContainer.insert(name, variable, variablesContainer.count());
+  };
+
+  componentDidMount() {
+    const { variablesContainer, objectVariables } = this.props;
+    if (objectVariables){
```

Comment:
To avoid more confusion, let's call the variables that are inherited "inheritedVariablesContainer" (please use "Container", it's more consistent with "variablesContainer") or "parentVariablesContainer" (because these variables are coming from the "parent", which is the object).

This may we'll have an editor that is reusable for other things, if one day variables can be inherited from something else ;)